modin/engines/ray/pandas_on_ray/frame/partition.py

```diff
@@ -189,6 +192,25 @@ def width(self):
                 handle_ray_task_error(e)
         return self._width_cache
 
+    def ip(self):
+        if self._ip_cache is None:
+            if len(self.call_queue):
+                self.drain_call_queue()
+            # Since `ray.wait` returns a tuple of completed and not completed object refs as lists
+            # we need to retrive only completed object ref via `[0][0]` indexing.
+            oid = ray.wait([self.oid])[0][0]
```

Comment:
I don't think we want to block on each IP address, it will force everything to block before it can move to the next partition. This combined with the `drain_call_queue` means that we will be operating single-threaded.

Reply:
I see these two possible cases:
1) We can notify a user as follows, for example, if you are going to use IPs, please note, that getting ip for each of the partitions is a blocking call.
2) We can return back to getting node ip addresses from remote functions taking into account the issue related to in-process store. `unwrap_partitions` would return a list of tuples of `ray.ObjectRefs` to ips and partitions, respectively. And `map_partitions_to_ips` would return a dictionary of ip -> list of partitions like `{str: [ObjectRef, ...], ...}` where we would materialize all of ips. For this `ip` property looks unnecessary.

Your thoughts?

Reply:
Agreed with the 2nd clause.